### DISCUSSION TITLE
Availability of dim for boundary conditions

### DISCUSSION THREAD

**Discussion by turbulentphil** [Nek5000/nekRS#573] [Q&A] [answered]
Dear Neks,

is there any way to retrieve the number of dimensions `dim` inside the .oudf boundary condition functions (`velocityDirichletConditions`, `scalarDirichletConditions`, `scalarNeumannConditions`)? 
I have consulted `bcData.h` but it seems to me that this does not contain the required information. Is there maybe more information (than just that from `bcData`) accessible from within these functions?

Best regards,
Philipp

**Comment by stgeke** [comment]
What do you mean by "number of dimensions"?

**Reply by turbulentphil** [reply]
I mean the dimensionality of the problem, 2 or 3.

**Comment by stgeke** [comment, marked as answer]
nekRS only supports 3D.On 30 Jun 2024, at 21:26, turbulentphil ***@***.***> wrote:﻿
I mean the dimensionality of the problem, 2 or 3.
